worker/logger/logger.go

```diff
@@ -20,16 +20,18 @@ var log = loggo.GetLogger("juju.worker.logger")
 type Logger struct {
 	api            *logger.State
 	agentConfig    agent.Config
+	updateCallback func(string) error
 	lastConfig     string
 	configOverride string
 }
 
 // NewLogger returns a worker.Worker that uses the notify watcher returned
 // from the setup.
-func NewLogger(api *logger.State, agentConfig agent.Config) (worker.Worker, error) {
+func NewLogger(api *logger.State, agentConfig agent.Config, updateCallback func(string) error) (worker.Worker, error) {
 	logger := &Logger{
 		api:            api,
 		agentConfig:    agentConfig,
```

Comment:
Not your change but... the only thing this appears to get used for is the agent tag, so perhaps just store that instead of holding on to the config.